libkbfs/md_journal.go

```diff
@@ -652,12 +679,36 @@ func (j *mdJournal) put(
 
 	mStatus := rmd.MergedStatus()
 
-	if (mStatus == Unmerged) && (rmd.BID() == NullBranchID) {
-		j.log.CDebugf(
-			ctx, "Changing branch ID to %s and prev root to %s for MD for TLF=%s with rev=%s",
-			lastBranchID, lastMdID, rmd.TlfID(), rmd.Revision(), rmd.BID())
-		rmd.SetBranchID(lastBranchID)
-		rmd.SetPrevRoot(lastMdID)
+	if mStatus == Unmerged {
```

Comment:
The logic in this function is getting quite hairy, but I can't see any way to simplify it. Any suggestions?